### PR TITLE
Fix leg annotation calculation

### PR DESCRIPTION
**Pull request by boldtrn**
Currently there are a few minor bugs when calculating the leg annotations. 

When you go offroute and calculate a new route, the leg annotation is not reset, which will lead to wrong `distanceIndex` in [NavigationHelper](https://github.com/boldtrn/maplibre-navigation-android/blob/0f8cf44e14c198d0bc0c31595aa4274005ae90e8/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelper.java#L523).

The same happens when you progress from legindex N to N+1. 

Additionally, there is a bug when you start a fresh navigation, that the route progress is calculated twice. In the first calculation the step distance remaining was missing, which lead to an incorrect legindex. We removed the first route progress calculation and made sure that the route progress is not required before that anymore.

### CODE REVIEW

**Comment by Fabi755** [comment]
Is it possible to add a test case for this?

**Comment by boldtrn** [comment]
Thanks @Fabi755, yes, we just found a few other things as well. We created tests and changelog entries for this and this PR should be ready for review. I will update the description above.

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelper.java [hunk 1]
```diff
@@ -526,7 +526,7 @@ private static int findAnnotationIndex(CurrentLegAnnotation currentLegAnnotation
     for (int i = distanceIndex; i < legDistances.size(); i++) {
       Double distance = legDistances.get(i);
       annotationDistancesTraveled += distance;
-      if (annotationDistancesTraveled > distanceTraveled) {
+      if (annotationDistancesTraveled > distanceTraveled || i == legDistances.size() -1 ) {
```
Comment (Fabi755) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelper.java:529]: What is here fixed? It's possible that `return INDEX_ZERO` in `line 535` not reachable anymore?

Is that covered by a test case?

Comment (boldtrn) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelper.java:529]: Correct, `INDEX_ZERO` should not be returned anymore. We left it untouched for historic reasons. 

`distanceTraveled >= annotationDistancesTraveled` can happen if you are at the end of leg. Before we returned index zero, which is wrong. Index 0 is the start of leg. If you have travelled the whole leg, we need to return the last index.

We actually found this bug when writing unit tests, so this is covered in a unit test implicitly.

Comment (Fabi755) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelper.java:529]: Nice. Sounds good for ne 💪

**Review by Fabi755 — APPROVED** [review]

**Pull request merged**